compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -758,8 +774,59 @@ public void visit(BLangTableConstructorExpr tableConstructorExpr) {
         }
     }
 
-    private boolean validateTableType(BTableType tableType, DiagnosticPos pos,
-                                      List<BLangRecordLiteral> recordLiterals) {
+    private BType inferTableMemberType(List<BType> memTypes, BLangTableConstructorExpr tableConstructorExpr) {
```

Comment:
I have these types defined

```ballerina
type Person record {
    readonly string name;
    readonly int age;
};

type Customer record {
    readonly int id;
    readonly string name;
    string lname;
};

```
And I have this table.

```ballerina
var tab = table key(name) [
    { name: "AAA", age: 31 },
    { name: "BBB", age: 34 },
    { id: 13 , name: "Foo", lname: "QWER" }
    ];
```

According to the PR table constraint will be one record type with required fields and some optional fields.
But shouldnt it be `table<Person|Customer>` ?